### PR TITLE
fix region may be missed

### LINKED ISSUE #2368: Table not found when TiDB has more than 10,000 tables in a DB
## Enhancement
When TiDB has more than 10,000 tables in one Database, TiSpark may throw exceptions like `Table not found`.

This problem may be because of the large schema, if so, we need to clarify how large the schema will lead to the problem. 

### Verison
TiDB 5.4
Spark 3.1.3
TiSpark 2.5

### PR DESCRIPTION
**Pull request by TrafalgarRicardoLu**
### What problem does this PR solve? 
close #2368 
When TiDB has more than 10,000 tables in one Database, TiSpark may throw Table not found  exceptions.

### What is changed and how it works?
The problem reported by the issue is caused by not timely update of region.

When there have stale data in region cache, the iterator will use end key in stale region to scan regions, and it may miss some new regions. And if we try to read the table in new region, table not found  exceptions will be thrown.

In fact, RegionStoreClient has already gotten the new region, but doesn't send it to ConcreteScanIterator. So we add the code to update region.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ConcreteScanIterator.java [hunk 1]
```diff
@@ -66,6 +66,7 @@ TiRegion loadCurrentRegionToCache() throws GrpcException {
           try {
             int scanSize = Math.min(limit, conf.getScanBatchSize());
             currentCache = client.scan(backOffer, startKey, scanSize, version);
+            region = client.getRegion();
```
Comment (xuanyu66) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ConcreteScanIterator.java:69]: Plz add some comment on the side effect of scan and why we update here

**Review by zhangyangyu — APPROVED** [review]

**Review by Daemonxiao — APPROVED** [review]

**Review by xuanyu66 — APPROVED** [review]

**Comment by shiyuhang0** [comment]
LGTM

**Pull request merged**